crates/matrix-sdk/src/widget/messages/event.rs

```diff
@@ -0,0 +1,79 @@
+use ruma::events::{MessageLikeEventType, StateEventType, TimelineEventType};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+use crate::widget::EventFilter;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct MatrixEvent {
+    #[serde(flatten)]
+    pub event_type: EventType,
+    pub sender: String,
+    pub event_id: String,
+    pub room_id: String,
+    pub origin_server_ts: u32,
+    pub unsigned: Unsigned,
+    // TODO: Really not sure about this one, I feel like it should be a `AnyTimelineEvent` or
+    // something like this. But @toger5 implemented it with this type originally, so let's
+    // clarify it better.
+    pub content: Value,
+}
```

Comment:
I hope we can replace this whole type with AnyTimelineEvent. The only thing that makes me wonder/hesitate is that AnyTimelineEvent does not impl Serialize (only Deserialize as far as I can tell). But maybe there is a way. `Raw<AnyTimelineEvent>`?

Reply:
Yes, `Raw<AnyTimelineEvent>` (or likely `Raw<AnySyncTimelineEvent>`) sounds like the right event type to pass around.